accounts/abi/bind/backends/simulated.go

```diff
@@ -589,6 +590,172 @@ func (b *SimulatedBackend) EstimateGas(ctx context.Context, call ethereum.CallMs
 	return hi, nil
 }
 
+// EstimateGasReviveState executes the requested code against the currently pending block/state and
+// returns the used amount of gas. If the execution fails due to expired state, it will revive the state
+// and try again.
+func (b *SimulatedBackend) EstimateGasReviveState(ctx context.Context, call ethereum.CallMsg) (uint64, error) {
```

Comment:
1. how about changing a new func name, may `EstimateGasAndReviveWitness`?
2. the generated witnessList could return back to user?